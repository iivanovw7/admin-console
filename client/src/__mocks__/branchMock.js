export const BRANCH = {
  address: '938 Barby Way',
  email: 'mbrazier0@amazon.co.uk',
  fax: '+86 829 425 7497',
  information: '1',
  name: 'San Mart√≠n de los Andes',
  phone: '+54 275 474 9318',
  status: true,
  __v: 0,
  _id: '5c8f96b9cece9911f01dd092'
};