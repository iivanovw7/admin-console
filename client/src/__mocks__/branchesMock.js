export const BRANCHES = {
  page: 1,
  limit: 7,
  pages: 4,
  results: 25,
  output: [
    {
      address: '938 Barby Way',
      email: 'mbrazier0@amazon.co.uk',
      fax: '+86 829 425 7497',
      information: '1',
      name: 'San Mart√≠n de los Andes',
      phone: '+54 275 474 9318',
      status: true,
      __v: 0,
      _id: '5c8f96b9cece9911f01dd092'
    },
    {
      address: '903 Dottie Street',
      email: 'inerger1@businesswire.com',
      fax: '+86 230 609 6364',
      information: 't',
      name: 'Dingzhai',
      phone: '+86 103 802 9655',
      status: true,
      __v: 0,
      _id: '5c8f96b9cece9911f01dd093'
    },
    {
      status:false,
      _id:"5c8f96b9cece9911f01dd0a0",
      name:"Cantao-an",
      email:"atroye@dailymail.co.uk",
      phone:"+63 269 360 0515",
      fax:"+86 713 306 1827",
      address:"7827 Luster Point",
    }
  ]
};
