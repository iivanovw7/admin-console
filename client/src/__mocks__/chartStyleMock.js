export const style = {
  type: "PieChart",
  id: 2
};
