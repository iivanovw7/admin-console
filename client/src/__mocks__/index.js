export * from './branchesMock';
export * from './branchMock';
export * from './userMock';
export * from './usersMock';
export * from './groupMock';
export * from './groupsMock';
export * from './roleMock';
export * from './rolesMock';
export * from './ticketMock';
export * from './ticketsMock';
export * from './messageMocks';
export * from './messagesMocks';
export * from './statisticsMock';
