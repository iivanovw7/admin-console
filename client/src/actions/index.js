export * from './authActions';
export * from './branchesActions';
export * from './groupsActions';
export * from './rolesActions';
export * from './usersActions';
export * from './ticketsActions';
export * from './messagesActions';
export * from './statisticsActions';