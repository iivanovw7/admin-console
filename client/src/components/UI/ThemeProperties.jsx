import blue from '@material-ui/core/colors/blue';
import { fade } from '@material-ui/core/styles/colorManipulator';
import { createMuiTheme } from '@material-ui/core/styles';

const drawerWidth = 240;

//Exporting material-ui main themeProperties configuration
export const themeProperties = createMuiTheme({
  typography: {
    useNextVariants: true
  },
  mixins: {
    toolbar: {
      textAlign: 'center',
      minHeight: '64px',
      justifyContent: 'space-between'
    }
  },
  palette: {
    primary: blue,
    secondary: {
      main: '#ffffff'
    }
  },
  status: {
    danger: 'orange'
  }
});

//Exporting material-ui main styles
export const NavigationStyles = theme => ({
  active: {
    backgroundColor: 'rgba(0, 0, 0, 0.10)'
  },
  root: {
    display: 'flex'
  },
  drawer: {
    [theme.breakpoints.up('sm')]: {
      width: drawerWidth,
      flexShrink: 0
    }
  },
  appBar: {
    marginLeft: drawerWidth,
    [theme.breakpoints.up('sm')]: {
      width: `calc(100% - ${drawerWidth}px)`
    }
  },
  menuButton: {
    marginRight: 20,
    [theme.breakpoints.up('sm')]: {
      display: 'none'
    }
  },
  toolbar: theme.mixins.toolbar,
  toolbarUserName: {
    [theme.breakpoints.down('xs')]: {
      display: 'none'
    }
  },
  drawerPaper: {
    width: drawerWidth
  },
  content: {
    flexGrow: 1,
    padding: theme.spacing.unit * 3
  },
  search: {
    position: 'relative',
    borderRadius: theme.shape.borderRadius,
    backgroundColor: fade(theme.palette.common.white, 0.15),
    '&:hover': {
      backgroundColor: fade(theme.palette.common.white, 0.25)
    },
    marginRight: theme.spacing.unit * 2,
    marginLeft: 0,
    width: '100%',
    [theme.breakpoints.up('sm')]: {
      marginLeft: theme.spacing.unit * 3,
      width: 'auto'
    }
  },
  searchIcon: {
    width: theme.spacing.unit * 9,
    height: '100%',
    position: 'absolute',
    pointerEvents: 'none',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center'
  },
  inputRoot: {
    color: 'inherit',
    width: '100%'
  },
  inputInput: {
    paddingTop: theme.spacing.unit,
    paddingRight: theme.spacing.unit,
    paddingBottom: theme.spacing.unit,
    paddingLeft: theme.spacing.unit * 10,
    transition: theme.transitions.create('width'),
    width: '100%',
    [theme.breakpoints.up('md')]: {
      width: 200
    }
  }
});

export const LoginFormStyles = theme => ({

  [theme.breakpoints.down('sm')]: {
    loginFormContainer: {
      boxShadow: 'none',
      height: '100%',
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'center'
    }
  },
  loginWrapper: {
    maxWidth: '500px',
    margin: '0 auto',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    height: '100%'
  },
  loginFormContainer: {
    padding: theme.spacing.unit,
    maxWidth: '500px'
  },
  formPaper: {
    margin: theme.spacing.unit * 2
  },
  paddingTop: theme.spacing.unit * 2,
  paddingBottom: theme.spacing.unit * 2

});

//Exporting material-ui main styles
export const Wrapper = theme => ({
  root: {
    display: 'flex'
  },
  toolbar: theme.mixins.toolbar,
  contentSingle: {
    flexGrow: 1,
    padding: theme.spacing.unit * 3,
    marginTop: '64px',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
    height: '100%',
  },
  contentList: {
    flexGrow: 1,
    padding: theme.spacing.unit * 3,
    marginTop: '64px',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
    [theme.breakpoints.up('sm')]: {
      height: '90vh'
    },
    [theme.breakpoints.down('sm')]: {
      height: '100%',
    }
  },
  formRoot: {
    display: 'flex',
    flexWrap: 'wrap',

    buttonWrapper: {
      display: 'flex',
      alignItems: 'center'
    }
  },
  fromLimitSelector: {
    display: 'flex',
    flexWrap: 'wrap',
    [theme.breakpoints.down('sm')]: {
      display: 'none'
    }
  },
  formControl: {
    margin: theme.spacing.unit,
    minWidth: 120,
    maxWidth: 300
  },
  controlsContainer: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingRight: 10,
    paddingLeft: 10
  },
  selectorsContainer: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    paddingRight: 10,
    paddingLeft: 10
  }
});

export const Container = theme => ({
  root: {
    width: '100%',
    marginTop: theme.spacing.unit * 3,
    overflowX: 'auto',
    display: 'flex',
    justifyContent: 'center',
    overflow: 'hidden'
  },
  tables: {
    minWidth: 200,
    width: '100%'
  },
  tableCell: {
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5
  },
  rowClass: {
    '&:hover': {
      backgroundColor: fade(theme.palette.common.black, 0.1),
      cursor: 'pointer'
    },
    [theme.breakpoints.up('sm')]: {
      height: '48px',
    },
    [theme.breakpoints.down('sm')]: {
      height: '100%',
    },
  },
  nameCell: {
    paddingRight: 5,
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5,

    [theme.breakpoints.up('sm')]: {
      width: '30%',
    },

    [theme.breakpoints.down('sm')]: {
      width: '60%',
    }

  },
  addressCell: {
    paddingRight: 5,
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5,
    width: '100%',
    [theme.breakpoints.down('sm')]: {
      display: 'none'
    }
  },
  employeesCell: {
    paddingRight: 5,
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5,
    width: '30%',
    [theme.breakpoints.down('sm')]: {
      display: 'none'
    }
  },
  branchPaper: {
    width: '100%',
    maxWidth: 600,
    margin: theme.spacing.unit * 2
  }
});

export const Notification = theme => ({
  close: {
    padding: theme.spacing.unit / 2
  }
});

export const Buttons = theme => ({
  formRoot: {
    display: 'flex',
    flexWrap: 'wrap'
  },
  [theme.breakpoints.down('sm')]: {
    mobile: {
      display: 'flex'
    },
    desktop: {
      display: 'none'
    }
  },
  [theme.breakpoints.up('sm')]: {
    mobile: {
      display: 'none'
    },
    desktop: {
      display: 'flex'
    }
  }
});

