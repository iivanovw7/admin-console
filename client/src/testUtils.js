import checkPropTypes from 'check-prop-types';
import { createStore } from 'redux';
import rootReducer from './reducers';

/**
 * @param {object} initialState - initial state to store
 * @function storeFactory
 * @returns {Store} Redux store
 */
export const storeFactory = initialState => {
  return createStore(rootReducer, initialState);
};

/**
 * Returns node(s) with the given test attributes
 * @param {object} wrapper - shallow wrapper
 * @param {string} val - data-test attribute to find
 * @returns {wrapper}
 */
export const findByTestAttr = (wrapper, val) => {
  return wrapper.find(`[data-test='${val}']`);
};