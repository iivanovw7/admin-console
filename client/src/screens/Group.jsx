import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { Paper } from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';
import { Wrapper } from '../components/UI/ThemeProperties';
import { getSingleGroup } from '../actions/groups';
import GroupContainer from '../components/Group/GroupContainer';

const Group = props => {
  const { classes, history } = props;
  const group = props.groups.group;

  return (
    <main className={classes.contentSingle}>
      <Paper className={classes.controlsContainer}>
        <div className={classes.selectorsContainer}>
          <h2>Create/Edit group</h2>
        </div>
      </Paper>
      {
        (group) ?
          (<GroupContainer history={history}/>) : (<p>Loading...</p>)
      }
    </main>
  );
};

Group.propTypes = {
  classes: PropTypes.object.isRequired,
  theme: PropTypes.object.isRequired
};

function mapStateToProps(state) {
  return {
    groups: state.groups
  };
}

export default connect(mapStateToProps, { getSingleGroup })(withStyles(Wrapper, { withTheme: true })(withRouter(Group)));