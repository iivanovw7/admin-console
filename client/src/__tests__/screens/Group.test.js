import { withStyles } from '@material-ui/core';
import { createMount, createShallow } from '@material-ui/core/test-utils';
import { createMemoryHistory } from 'history';
import jest from 'jest-mock';
import React from 'react';
import { connect, Provider } from 'react-redux';
import { MemoryRouter, withRouter } from 'react-router-dom';
import { getSingleGroup } from '../../actions';
import { Wrapper } from '../../components/UI/ThemeProperties';
import Group from '../../screens/Group';
import { storeFactory } from '../../testUtils';
import * as mocks from './../../__mocks__/';

const history = createMemoryHistory('/dashboard');

const setup = (initialState = {}) => {
  let wrapper;
  let shallowWithDive;
  let mount;
  let shallow;
  let store = storeFactory(initialState);

  const Composition = connect(null, { getSingleGroup })(withStyles(Wrapper, { withTheme: true })(withRouter(Group)));
  const props = {
    history: history,
    groups: {
      group: mocks.GROUP
    }
  };

  beforeEach(() => {
    shallow = createShallow();
    shallowWithDive = createShallow({ dive: true });
    mount = createMount();
  });

  afterEach(() => {
    mount.cleanUp();
  });

  /**
   *  Renders group main screen form with styles,
   *  Checks if component rendered properly.
   */
  describe('Group screen', () => {
    it('Should render HOC with styles properly', () => {
      const mockFunc = jest.fn();
      wrapper = mount(
        <Provider store={store}>
          <MemoryRouter>
            <Composition {...props} dispatch={mockFunc}/>
          </MemoryRouter>
        </Provider>
      );
      expect(wrapper.html()).not.toBe(null);
      expect(wrapper.html()).not.toBe(undefined);
      expect(wrapper.length).toEqual(1);
      expect(wrapper.find('h2').text()).toBe('Create/Edit group');
      expect(wrapper.html()).toMatchSnapshot();
    });
  });
};

setup();



