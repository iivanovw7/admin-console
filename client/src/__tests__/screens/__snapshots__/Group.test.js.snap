// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Group screen Should render HOC with styles properly 1`] = `"<main class=\\"withRouter-Group--contentSingle-18 withRouter-Connect-WithStyles-withRouter-Group-----contentSingle-4\\"><div class=\\"MuiPaper-root-29 MuiPaper-elevation2-33 MuiPaper-rounded-30 withRouter-Group--titleContainer-24 withRouter-Connect-WithStyles-withRouter-Group-----titleContainer-10\\"><div class=\\"withRouter-Group--selectorsContainer-26 withRouter-Connect-WithStyles-withRouter-Group-----selectorsContainer-12\\"><h2>Create/Edit group</h2></div></div><div class=\\"MuiPaper-root-29 MuiPaper-elevation2-33 MuiPaper-rounded-30 ReduxForm-root-56\\"><form class=\\"ReduxForm-branchPaper-68\\"><br><div class=\\"MuiGrid-container-82 MuiGrid-spacing-xs-16-105 MuiGrid-align-items-xs-flex-end-92\\"><div class=\\"MuiGrid-item-83 MuiGrid-grid-xs-true-110 MuiGrid-grid-sm-true-124 MuiGrid-grid-md-true-138\\"><div class=\\"MuiFormControl-root-179 MuiFormControl-fullWidth-182\\"><label class=\\"MuiFormLabel-root-194 MuiFormLabel-required-199 MuiInputLabel-required-187 MuiInputLabel-root-183 MuiInputLabel-formControl-188 MuiInputLabel-animated-191 MuiInputLabel-outlined-193\\" data-shrink=\\"false\\" for=\\"name\\">name<span class=\\"MuiFormLabel-asterisk-200\\"> *</span></label><div class=\\"MuiInputBase-root-214 MuiOutlinedInput-root-201 MuiInputBase-fullWidth-223 MuiInputBase-formControl-215\\"><fieldset aria-hidden=\\"true\\" style=\\"padding-left: 8px;\\" class=\\"MuiPrivateNotchedOutline-root-231 MuiOutlinedInput-notchedOutline-208\\"><legend class=\\"MuiPrivateNotchedOutline-legend-232\\" style=\\"width: 0.01px;\\"><span>​</span></legend></fieldset><input aria-invalid=\\"false\\" aria-describedby=\\"name-helper-text\\" class=\\"MuiInputBase-input-224 MuiOutlinedInput-input-209\\" id=\\"name\\" name=\\"name\\" required=\\"\\" rows=\\"1\\" type=\\"text\\" value=\\"\\"></div><p class=\\"MuiFormHelperText-root-233 MuiFormHelperText-contained-237 MuiFormHelperText-required-240\\" id=\\"name-helper-text\\"> </p></div></div></div><div class=\\"MuiGrid-container-82 MuiGrid-spacing-xs-16-105 MuiGrid-align-items-xs-flex-end-92\\"><div class=\\"MuiGrid-item-83 MuiGrid-grid-xs-true-110 MuiGrid-grid-sm-true-124 MuiGrid-grid-md-true-138\\"><div class=\\"MuiFormControl-root-179 MuiFormControl-fullWidth-182\\"><label class=\\"MuiFormLabel-root-194 MuiFormLabel-required-199 MuiInputLabel-required-187 MuiInputLabel-root-183 MuiInputLabel-formControl-188 MuiInputLabel-animated-191 MuiInputLabel-outlined-193\\" data-shrink=\\"false\\" for=\\"description\\">description<span class=\\"MuiFormLabel-asterisk-200\\"> *</span></label><div class=\\"MuiInputBase-root-214 MuiOutlinedInput-root-201 MuiInputBase-fullWidth-223 MuiInputBase-formControl-215 MuiInputBase-multiline-222 MuiOutlinedInput-multiline-207\\"><fieldset aria-hidden=\\"true\\" style=\\"padding-left: 8px;\\" class=\\"MuiPrivateNotchedOutline-root-231 MuiOutlinedInput-notchedOutline-208\\"><legend class=\\"MuiPrivateNotchedOutline-legend-232\\" style=\\"width: 0.01px;\\"><span>​</span></legend></fieldset><div class=\\"MuiPrivateTextarea-root-241\\"><textarea aria-hidden=\\"true\\" class=\\"MuiPrivateTextarea-textarea-242 MuiPrivateTextarea-shadow-243\\" readonly=\\"\\" rows=\\"1\\" tabindex=\\"-1\\"></textarea><textarea aria-hidden=\\"true\\" class=\\"MuiPrivateTextarea-textarea-242 MuiPrivateTextarea-shadow-243\\" readonly=\\"\\" rows=\\"4\\" tabindex=\\"-1\\"></textarea><textarea rows=\\"4\\" class=\\"MuiPrivateTextarea-textarea-242 MuiInputBase-input-224 MuiOutlinedInput-input-209 MuiInputBase-inputMultiline-226 MuiOutlinedInput-inputMultiline-211\\" style=\\"height: 19px;\\" aria-invalid=\\"false\\" aria-describedby=\\"description-helper-text\\" id=\\"description\\" name=\\"description\\" required=\\"\\"></textarea></div></div><p class=\\"MuiFormHelperText-root-233 MuiFormHelperText-contained-237 MuiFormHelperText-required-240\\" id=\\"description-helper-text\\"> </p></div></div></div><div class=\\"MuiGrid-container-82 MuiGrid-spacing-xs-16-105 MuiGrid-align-items-xs-flex-end-92\\"><div class=\\"MuiGrid-item-83 MuiGrid-grid-xs-true-110 MuiGrid-grid-sm-true-124 MuiGrid-grid-md-true-138\\"><label class=\\"MuiFormControlLabel-root-244\\"><span class=\\"MuiButtonBase-root-266 MuiIconButton-root-260 MuiPrivateSwitchBase-root-256 MuiCheckbox-root-250 MuiCheckbox-colorPrimary-254\\"><span class=\\"MuiIconButton-label-265\\"><svg class=\\"MuiSvgIcon-root-269\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z\\"></path></svg><input class=\\"MuiPrivateSwitchBase-input-259\\" id=\\"permissions\\" type=\\"checkbox\\" data-indeterminate=\\"false\\" value=\\"\\"></span><span class=\\"MuiTouchRipple-root-340\\"></span></span><span class=\\"MuiTypography-root-278 MuiTypography-body1-287 MuiFormControlLabel-label-249\\">Available for permissions</span></label></div></div><div class=\\"MuiGrid-container-82 MuiGrid-spacing-xs-16-105 MuiGrid-align-items-xs-flex-end-92\\"><div class=\\"MuiGrid-item-83 MuiGrid-grid-xs-true-110 MuiGrid-grid-sm-true-124 MuiGrid-grid-md-true-138\\"><label class=\\"MuiFormControlLabel-root-244\\"><span class=\\"MuiButtonBase-root-266 MuiIconButton-root-260 MuiPrivateSwitchBase-root-256 MuiCheckbox-root-250 MuiCheckbox-colorPrimary-254\\"><span class=\\"MuiIconButton-label-265\\"><svg class=\\"MuiSvgIcon-root-269\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z\\"></path></svg><input class=\\"MuiPrivateSwitchBase-input-259\\" id=\\"status\\" type=\\"checkbox\\" data-indeterminate=\\"false\\" value=\\"\\"></span><span class=\\"MuiTouchRipple-root-340\\"></span></span><span class=\\"MuiTypography-root-278 MuiTypography-body1-287 MuiFormControlLabel-label-249\\">Active</span></label></div></div><div class=\\"MuiGrid-container-82 MuiGrid-justify-xs-flex-end-100\\" style=\\"margin-top: 10px;\\"><button class=\\"MuiButtonBase-root-266 MuiButton-root-314 MuiButton-contained-325 MuiButton-containedPrimary-326 MuiButton-raised-328 MuiButton-raisedPrimary-329\\" tabindex=\\"0\\" type=\\"button\\" style=\\"text-transform: none; margin: 5px;\\"><span class=\\"MuiButton-label-315\\">CANCEL</span><span class=\\"MuiTouchRipple-root-340\\"></span></button><button class=\\"MuiButtonBase-root-266 MuiButton-root-314 MuiButton-contained-325 MuiButton-containedPrimary-326 MuiButton-raised-328 MuiButton-raisedPrimary-329\\" tabindex=\\"0\\" type=\\"submit\\" style=\\"text-transform: none; margin: 5px;\\"><span class=\\"MuiButton-label-315\\">SAVE</span><span class=\\"MuiTouchRipple-root-340\\"></span></button></div></form></div></main>"`;
